### DISCUSSION TITLE
Space-x and space-y utilities should exclude `sr-only` class

### DISCUSSION THREAD

**Discussion by brycekunkel** [tailwindlabs/tailwindcss#15408] [Ideas]
Right now, the `space-x` and `space-y` utilities automatically do not apply margin to elements designated as `:hidden`. Since margin is a visual change on the page, the utility could be improved by not applying margin to elements with the `sr-only` class. The resulting rule might look something like this: 
```css
.space-x-1 > :not([hidden]):not(.sr-only) ~ :not([hidden]):not(.sr-only) {
  ...
}
```
Even better if the utility also supported other classes that remove the item from visibility on the page, like `.hidden` and `.invisible`.

**Comment by adamwathan** [comment]
This is super, super hard unfortunately because of stuff like `md:sr-only`. Best bet is to just use the `gap-*` utilities instead where the browser handles all of this stuff for you.

**Comment by brycekunkel** [comment]
Definitely make sense. Thanks for responding - I'll close this one out.
